src/Components/AppLifecycleTable/appLifecycleTable.tsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import { Table, Tbody, Tr, Td } from '@patternfly/react-table';
+import Moment from 'moment';
+
+interface dataAppLifeCycleChanges {
+    data: AppLifecycleChanges[];
+
+}
+
+interface AppLifecycleChanges {
+    module_name: string;
+    rhel_major_version: number;
+    streams: Stream[];
+  }
+  
+interface Stream {
+    arch: string;
+    context: string;
+    description: string;
+    end_date: string;
+    name: string;
+    profiles: Profiles;
+    start_date: string;
+    stream: string;
+    version: string;
+  }
+  
+interface Profiles {
+    common: string[];
+  }
+
+const columnNames = {
+  name: 'Name',
+  release: 'Release',
+  release_date: 'Release Date',
+  retirement_date: 'Retirement Date',
+  systems: 'Systems'
+};
+
+export const appLifecycleTab: React.FunctionComponent<dataAppLifeCycleChanges> = (
+  {data}
+) => {
+    const rows = data.flatMap(repo => repo.streams).map(stream => (
+        <Tr key={stream.name}>
+          <Td style={{ paddingRight: '140px' }} dataLabel={columnNames.name}>{stream.name}</Td>
```

Comment:
```suggestion
          <Td style={{ paddingRight: '140px' }} dataLabel={columnNames.name}>{data.find((item) => item.module_name === stream.name)?.rhel_major_version}</Td>
```
^ this is what I have now, but still require some work (it's showing only 8 rhel major)

Reply:
Find is only going to return the first value that matches: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

You're going to want to try something like filter, I think - that'll get you all values that match a given criteria rather than the first.

Reply:
UP: I had to combine the `filter` with `some`